koku/api/query_params.py

```diff
@@ -146,35 +186,33 @@ def _set_access(self, filter_key, access_key, raise_exception=True):
             elif access_list:
                 self.parameters["filter"][filter_key] = access_list
 
-    def _set_access_aws(self):
-        """Alter query parameters based on user access."""
-        self._set_access("account", "aws.account")
-
-    def _set_access_azure(self):
-        """Alter query parameters based on user access."""
-        self._set_access("subscription_guid", "azure.subscription_guid")
-
-    def _set_access_ocp(self):
-        """Alter query parameters based on user access."""
-        params = [("cluster", True), ("node", False), ("project", False)]
-        for name, exc in params:
-            self._set_access(name, f"openshift.{name}", raise_exception=exc)
-
-    def _set_access_ocp_aws(self):
+    def _set_access_ocp_all(self, provider, filter_key, access_key, raise_exception=True):
         """Alter query parameters based on user access."""
-        self._set_access_aws()
-        self._set_access_ocp()
+        access_list = self.access.get(access_key, {}).get("read", [])
+        access_filter_applied = False
+        if ReportQueryHandler.has_wildcard(access_list):
```

Comment:
We should accommodate the special case where both `AWS` and `Azure` have wildcards. When both (or all infrastructure/cloud providers in the future) have a wildcard, we can just not filter then.

Reply:
I was so adamant about this I commented twice.